pkg/controllers/customcluster_helper.go

```diff
@@ -413,3 +415,126 @@ func hasProvisionClusterInfo(phase v1alpha1.CustomClusterPhase) bool {
 	}
 	return false
 }
+
+// fetchProvisionedClusterKubeConfig fetch provisioned cluster’s kubeConfig file, and create a secret named "provisionedClusterKubeConfigSecretPrefix + customCluster.name" with the data of kube-config file.
+func (r *CustomClusterController) fetchProvisionedClusterKubeConfig(ctx context.Context, customCluster *v1alpha1.CustomCluster, customMachine *v1alpha1.CustomMachine) error {
+	remoteMachineSSHKey := customMachine.Spec.Master[0].SSHKey
+	controlPlaneHost := customMachine.Spec.Master[0].PublicIP
+
+	sshKeySecret, err := r.getSSHKeySecret(ctx, customMachine.Namespace, remoteMachineSSHKey.Name)
+	if err != nil {
+		return err
+	}
+
+	sshConfig, err := r.buildSSHClientConfig(sshKeySecret)
+	if err != nil {
+		return err
+	}
+
+	sftpClient, err := r.connectSFTPClient(controlPlaneHost+":22", sshConfig)
+	if err != nil {
+		return err
+	}
+	defer sftpClient.Close()
+
+	kubeConfigData, err := r.fetchRemoteKubeConfig(sftpClient, getProvisionedKubeConfigPath())
+	if err != nil {
+		return err
+	}
+
+	err = r.createKubeConfigSecret(ctx, generateProvisionedClusterSecretName(customCluster), customCluster.Namespace, kubeConfigData)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *CustomClusterController) getSSHKeySecret(ctx context.Context, namespace, name string) (*corev1.Secret, error) {
```

Comment:
```suggestion
func (c *CustomClusterController) getSSHKeySecret(ctx context.Context, namespace, name string) (*corev1.Secret, error) {
```

Reply:
The current code typically uses "r" instead of "c" to represent *CustomClusterController. If this is changed to "c", it may require modifications to many files.